src/adapter_test_helpers/sumo_basic_test_helper.erl

```diff
@@ -176,6 +177,24 @@ check_proper_dates(Config) ->
   Date = Module:birthdate(Person),
   ok.
 
+-spec persist_using_changeset(config()) -> ok.
+persist_using_changeset(Config) ->
+  {_, Name} = lists:keyfind(name, 1, Config),
+  Module = sumo_config:get_prop_value(Name, module),
+  Schema = Module:sumo_schema(),
+
+  [] = sumo:find_by(Name, [{name, <<"John">>}]),
+  [P1] = sumo:find_by(Name, [{name, <<"A">>}]),
+  Doc1 = sumo:new_doc(people, P1),
```

Comment:
Ok, I agree, what if we support both options: `sumo_changeset:cast/3` and `sumo_changeset:cast/4` ? And `sumo_changeset:cast/4` would use the existing one, something like this:

```erlang
cast(SchemaName, UserDoc, Params, Allowed) ->
  cast(sumo:new_doc(SchemaName, UserDoc), Params, Allowed).
```